gatsby/types.d.ts

```diff
@@ -0,0 +1,11 @@
+// Mock window.dataLayer, part of GTM. Global nav header relies on this
```

Comment:
Might as well copy the whole of the definition from https://github.com/nice-digital/bnf-gatsby/pull/25/files#diff-28c6813648154bffd3c879770c59235a7a406d5e7130516d6bfe2f0d3543b703R1-R15

Reply:
Ah ok, wasn't sure which bits were specific to the search.resultLoaded event.